### ISSUE TITLE
Login not working when deployed in Heroku

### ISSUE THREAD

**Issue by UO269911** [Arquisoft/radarin_en2a#134]
```
2021-04-12T07:22:10.960397+00:00 app[web.1]: (node:3) UnhandledPromiseRejectionWarning: OPError: invalid_grant (Mismatching redirect uri)

2021-04-12T07:22:10.960408+00:00 app[web.1]:     at processResponse (/usr/src/app/node_modules/openid-client/lib/helpers/process_response.js:45:13)

2021-04-12T07:22:10.960409+00:00 app[web.1]:     at Client.grant (/usr/src/app/node_modules/openid-client/lib/client.js:1237:26)

2021-04-12T07:22:10.960411+00:00 app[web.1]:     at runMicrotasks (<anonymous>)

2021-04-12T07:22:10.960412+00:00 app[web.1]:     at processTicksAndRejections (internal/process/task_queues.js:94:5)

2021-04-12T07:22:10.960412+00:00 app[web.1]:     at async Client.callback (/usr/src/app/node_modules/openid-client/lib/client.js:460:24)

2021-04-12T07:22:10.960414+00:00 app[web.1]:     at async AuthCodeRedirectHandler.handle (/usr/src/app/node_modules/@inrupt/solid-client-authn-node/dist/login/oidc/redirectHandler/AuthCodeRedirectHandler.js:79:24)

2021-04-12T07:22:10.960415+00:00 app[web.1]:     at async ClientAuthentication.handleIncomingRedirect (/usr/src/app/node_modules/@inrupt/solid-client-authn-node/dist/ClientAuthentication.js:71:34)

2021-04-12T07:22:10.960416+00:00 app[web.1]:     at async Session.handleIncomingRedirect (/usr/src/app/node_modules/@inrupt/solid-client-authn-node/dist/Session.js:45:35)

2021-04-12T07:22:10.960416+00:00 app[web.1]:     at async /usr/src/app/controllers/SessionController.js:44:26

2021-04-12T07:22:10.960426+00:00 app[web.1]: (node:3) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)

2021-04-12T07:22:40.494024+00:00 heroku[router]: at=error code=H12 desc="Request timeout" method=GET path="/api/session/login/redirect?sessionId=2803aec5-0be4-4335-8c0b-9e2aac777d1b&redirectUrl=https%3A%2F%2Fradarinen2awebapp.herokuapp.com%2F&code=592f036deed002655fec3b9c4d09c14c&state=NSZxPGq-RqGzG3GxQ4LiKCO2Z3qi-CmGMzFisgRHLTQ" host=radarinen2arestapi.herokuapp.com request_id=74f4a6b0-1efa-47e6-b532-86631302ea0c fwd="83.97.144.126" dyno=web.1 connect=1ms service=30000ms status=503 bytes=0 protocol=https
```

**Comment by UO269911** [comment]
Tried to fix in 7e684ffb78b76bb5a929a2d78f5ca3ef09f4eb7f by storing the final redirect URL in the restapi instead of sending it through the identity provider. Still same error.

**Comment by UO269911** [comment]
Fixed in 53499817d339fa74ddaf43c9e4a18d7eec54c143. 
```js
// When deployed in Heroku, it uses HTTPS but even though we specify https:// in the redirectUrl parameter in session.login, the identity provider redirects using HTTP
// so then session.handleIncomingRedirect sees that the URL is different since it has http:// instead of https:// and fails with `OPError: invalid_grant (Mismatching redirect uri)`.
// Workaround by replacing it back to https://.
```

For example, the initial `redirectUrl` in `/session/login` was:
```
https://radarinen2arestapi.herokuapp.com/api/session/login/redirect?sessionId=1234
```
But in `/session/login/redirect`, `getRequestFullUrl(req)` returned:
```
http://radarinen2arestapi.herokuapp.com/api/session/login/redirect?sessionId=1234&code=4567&state=8901
```
